src/sinks/socket.rs

```diff
@@ -276,6 +279,45 @@ mod test {
         }
     }
 
+    #[tokio::test]
+    async fn metrics_socket() {
```

Comment:
This added test won't be able to run on Windows. Do you mind gating it to run only on *nix? `#[cfg(target_family = "unix")]`

Reply:
Done. (Added `#cfg(unix)` which seem to be used elsewhere). Let's see if windows is happy.